controllers/component_controller.go

```diff
@@ -253,8 +253,22 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 	}
 
+	if component.Spec.Source.ImageSource != nil {
+
+		// TODO: would move this to the user's gitops repository under /.tekton
+		webhookURL, err := r.setupWebhookTriggeredImageBuilds(ctx, log, component)
+		if err != nil {
+			log.Error(err, "Unable to setup builds")
+		}
+		component.Status.Webhook = webhookURL
+		err = r.Client.Status().Update(ctx, &component)
+		if err != nil {
+			log.Error(err, "Unable to update Component with webhook URL")
+		}
```

Comment:
so i think we should do this only when the webhook returned is different.

Going by your previous comment [here](https://github.com/redhat-appstudio/application-service/pull/22#issuecomment-992723798), if there is already a webhook url created, we dont necessarily want to keep updating this webhook field everytime it goes thru reconcile 🤔

Reply:
That is correct.